Work around missing `akka-http` Scala 3 artifacts for tests

mkurz opened the pull request:
Running tests with Scala 3 needs to be done with a `-Dscala3Tests=true` flag.
This causes akka dependencies to use `cross(CrossVersion.for3Use2_13)`.  That's because we use `for3Use2_13` for akka-http as well. If we wouldn't apply `for3Use2_13` for akka now, akka-http would not be able to work with akka Scala 3 artifacts... In short: It sucks.
The flag also excludes some other dependencies were _3 and _2.13 are conflicting now because of the above.

The flag only should be applied when running tests. because for publishing we do want to publish all dependencies with `_3`.
I am working on some sbt plugin magic for the akka-http sbt plugin so it switches to `for3Use2_13` when using it with Scala 3 automatically. Not sure if it will work, but we will see.

mkurz commented:
BTW: I am testing all of that locally for now because it's not possible currently to build Scala 3 artifacts from the current main branch yet. However, there is just some specs2 magic left to fix and if that is done, all missing pieces fit together and we can push/merge a couple of PRs to make Scala 3 work on CI. We are really close and this is very exciting (for me at least :wink: )

mkurz commented on dev-mode/sbt-plugin/src/main/scala/play/sbt/Play.scala:
```diff
@@ -106,9 +106,44 @@ object PlayNettyServer extends AutoPlugin {
+         scala2Deps.flatMap(e => e._2._2.map(e._1 %% _ % e._2._1).map(_.cross(CrossVersion.for3Use2_13))).toSeq
+       } else {
+         Seq.empty
+       }),
```
excluding the dependencies and then including them again with the `for3Use2_13` workaround is the only approach that worked for me... I couldn't come up with something else. I thought `dependencyOverrides` might work, however with that you can only override versions of an exact "<organization>" %% "<groupId>". I was digging into the sbt and coursier code but there is no way to make a dependency switch the cross version (from _2.13 to _3).

BillyAutrey commented:
> > What do you think about that and the wording?
> 
> I found the message bit verbose, and there is no way to disable it. What about a 1 line message with a link to play documentation site?

I agree with this sentiment. The message is pretty verbose. You could probably get away with:

```
[info] Scala 3 requires additional dependencies to use PlayAkkaHttpServer.
[info] The following dependencies have automatically been imported as Scala 2 artifacts with cross-version support:
[info] 
[info] com.typesafe.akka %% akka-actor_2.13 % 2.6.20
[info] com.typesafe.akka %% akka-actor-typed_2.13 % 2.6.20
[info] com.typesafe.akka %% akka-slf4j_2.13 % 2.6.20
[info] com.typesafe.akka %% akka-serialization-jackson_2.13 % 2.6.20
[info] com.typesafe.akka %% akka-stream_2.13 % 2.6.20
[info] com.typesafe %% ssl-config-core_2.13 % 0.6.1
[info]
[info] Details at http://playframework.com/scala_3_compat_link
[info]
[info] com.fasterxml.jackson.module %% jackson-module-scala_2.13 % 2.14.2
```

You might even be able to get away with not providing the link, if we SEO the message. Though the link would be better to include.

mkurz commented:
> `[info] Scala 3 requires additional dependencies to use PlayAkkaHttpServer.`

They are not really additional, all of them get pulled in by the akka-http plugin normally, it's just that we switch them to be _2.13 instead of _3.
I removed the message for now and added it as todo to the Scala 3 migration guide (#11739). Probably there will be section anyway we can link to.

mkurz commented:
Merging. Good thing here is this all of this workarounds a conditional and in no way affect the Scala 2 build.

mkurz commented:
As part of #11804 I changed the text to point to the migration guide now:

```
[info] Using Scala 3 and the Akka HTTP server backend, but without native Scala 3 artifacts:
[info] https://www.playframework.com/documentation/latest/Scala3Migration#Using-Scala-3-with-Akka-HTTP-10.5-or-newer
```
